pandas/core/groupby/groupby.py

```diff
@@ -2220,26 +2220,28 @@ def _get_cythonized_result(
         base_func = getattr(libgroupby, how)
 
         for name, obj in self._iterate_slices():
+            values = obj._data._values
```

Comment:
Hmm I'm not sure relying on underlying block values is the best way to go about this. Is it possible to just work with the obj here instead?

Reply:
Sorry for my lack of understanding about underlying block values.
Should I leave the code using `obj.values` except for the last `result = algorithms.take_nd(obj.values, result)`?